spec/system/post_index_spec.rb

```diff
@@ -0,0 +1,98 @@
+RSpec.describe 'Post index page', type: :feature do
+  before(:each) do
+    @user = User.create(
+      name: 'Mohi',
+      bio: 'Aspiring FullStack Dev',
+      photo: 'https://via.placeholder.com/150',
+      posts_counter: 2
+    )
+
+    @user2 = User.create(
+      name: 'nati',
+      bio: 'FullStack Dev',
+      photo: 'https://via.placeholder.com/150',
+      posts_counter: 2
+    )
+
+    @post1 = Post.create(
+      title: 'Test',
+      text: 'First Post',
+      comments_counter: 0,
+      likes_counter: 0,
+      author: @user
+    )
+
+    @post2 = Post.create(
+      title: 'Test 2',
+      text: 'Second Post',
+      comments_counter: 0,
+      likes_counter: 0,
+      author: @user2
+    )
+
+    @comment1 = Comment.create(
+      text: 'First Comment',
+      post: @post1,
+      author: @user2
+    )
+
+    @comment2 = Comment.create(
+      text: 'Second Comment',
+      post: @post1,
+      author: @user2
+    )
+
+    @like1 = Like.create(
+      post: @post1,
+      author: @user2
+    )
+  end
+
+  describe 'the post index page' do
+    it 'displays user profile picture' do
+      visit user_posts_path(@user)
+      expect(page).to have_css("img[src*='https://via.placeholder.com/150']")
+    end
+
+    it 'displays the user username' do
+      visit user_posts_path(@user)
+      expect(page).to have_content('Mohi')
+    end
+
+    it 'displays number of posts the user has written' do
+      visit user_posts_path(@user)
+      expect(page).to have_content('Number of posts: 3')
+    end
+
+    it 'displays how many likes a post has' do
+      visit user_posts_path(@user)
+      expect(page).to have_content('likes: 1')
+    end
+
+    it 'displays some of the posts body' do
+      visit user_posts_path(@user)
+      expect(page).to have_content('First Post')
+    end
+
+    it 'displays posts title.' do
+      visit user_posts_path(@user)
+      expect(page).to have_content('Test')
+    end
+
+    it 'displays the first comments on a post.' do
+      visit user_posts_path(@user)
+      expect(page).to have_content('First Comment')
+    end
+
+    it 'section for pagination if there are more posts than fit on the view' do
+      visit user_posts_path(@user)
+      expect(page).to have_content('pagination')
+    end
+
+    it 'click on a post, it redirects to that post show page.' do
+      visit user_posts_path(@user)
+      click_link 'Test'
+      expect(page).to have_current_path(user_post_path(@user, @post1))
+    end
+  end
```

Comment:
- [ ]  As mentioned by the previous reviewer and for the project's requirements, kindly ensure that the index show page has all required tests as the following featured test(s) are missing:
   * Tests that asserts you can see how many comments a post has.

- [ ] Once you have fixed the above change, kindly ensure that all your tests are still passing as right now while testing your tests, I noticed that some of them are not passing instead this is the major requirement of this project. Below is the result picture of your tests.
![Screenshot from 2023-05-25 00-47-24](https://github.com/mohisa302/Blog-app/assets/83042742/db947d47-68db-4279-96eb-e491b038c015)

Reply:
I also checked the resting data base, and some different situations, but I didn't get an error.

Reply:
Hi @mohisa302 
Are there any changes you didn't push?

Reply:
Hey @tresorsawasawa, No actually. I just reseted the database now and added the test case you mentioned before.

Reply:
so what do you think I need to do to fix the problem?

Reply:
Hi @mohisa302 

If you have updated somethings, make sure all are pushed on github.

Reply:
Hey @tresorsawasawa , I pushed already all the changes:
**on branch fix**
**Your branch is up to date with 'origin/fix'.**

**nothing to commit, working tree clean**

Reply:
I'm working on another branch for another project but it doesn't impact fix branch.

Reply:
Hi @tresorsawasawa , I still not sure what is the problem with the tests. would you guide me?

Reply:
Dear @gemmen29 , I was wondering if I your side also there is an error with the tests. could you please let me know what is the error exactly if there is one?

Reply:
Answered on Slack